web/static/js/components/subjects/SubjectForm.jsx

```diff
@@ -0,0 +1,42 @@
+// @flow
+import React, { Component } from 'react'
+import TextField from 'react-md/lib/TextFields'
+import Button from 'react-md/lib/Buttons'
+import type {SubjectParams} from '../../types'
+
+class SubjectForm extends Component {
```

Comment:
Using Flowjs, we can type internal component state and props. Typing props is quite useful to surface little errors when using the components, while typing state helps ensuring that you're thinking of all the alternatives you need to take into account when rendering for example. Anyway this component is super clear, it's nice as is and it's not necessary to add types at this time. Just a suggestion for the future.